### PR TITLE
Closes #2713: Weave in "onboarding state" and split adapter items acc…

### LINKED ISSUE #2713: [Bug] Onboarding - Blank screen after tapping Sign in, Sign in
In the logs, I'm seeing the following:

```
FirefoxAccountStateMachine  D  Processing event Init for state Start. Next state is Start
                         D  Got invalid event Authenticate for state Start.
                         D  Ran 'Init' side-effects for state Start, got successive event AccountNotFound
                         D  Processing event AccountNotFound for state Start. Next state is NotAuthenticated
            AppServices  I  Using fxaclient_ffi_lib_name: fenix
          fxaclient_ffi  D  fxa_new
              GeckoView  W  No GeckoSession attached to this GeckoView instance. Call setSession to attach a GeckoSession to this instance.
```

Seems like state events are not being processed in the right order.

## Steps to reproduce
With a fresh Fenix install, click "Sign in" in the onboarding flow, then "sign in" again.

### Expected behavior
FxA sign-in page.

### Actual behavior
Blank screen. Going back, and tapping "sign in" again fixes the problem.

### Device information
* Android device: ?
* Fenix version: ?

### PR DESCRIPTION
**Pull request by grigoryk**
Missing UI customizations, probably will need new strings, and missing an account observer listener impl (to live in HomeFragment, I think).

cc @boek 

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

**Comment by grigoryk** [comment]
@boek do you reckon this is the right path for customizing onboarding behaviour for different account states? I need to work on other things for the next few days, so feel free to do whatever with this patch if it's helpful.

**Comment by boek** [comment]
@grigoryk yep! I'm wrapping it up now

**Comment by boek** [comment]
@ekager @sblatz this is ready to review now :+1:

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 1]
```diff
@@ -612,13 +623,24 @@ class HomeFragment : Fragment(), CoroutineScope {
     }
 
     private fun currentMode(): Mode = if (!onboarding.userHasBeenOnboarded()) {
-        Mode.Onboarding
+        // TODO monitor account state changes somewhere in this class via AccountObserver + `accountManager.register()`.
```
Comment (sblatz) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: Do we have a ticket filed for this?

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlComponent.kt [hunk 2]
```diff
@@ -64,10 +64,23 @@ fun List<Tab>.toSessionBundle(context: Context): MutableList<Session> {
 
     return sessionBundle
 }
+
+/**
+ * Describes various onboarding states.
+ */
+enum class OnboardingState {
+    // signed out, no account carried over from Fennec.
```
Comment (sblatz) [inline app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlComponent.kt]: nit: Let's capitalize these comments for consistency.

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlUIView.kt [hunk 3]
```diff
@@ -61,25 +61,45 @@ private fun privateModeAdapterItems(tabs: List<Tab>): List<AdapterItem> {
...
-    AdapterItem.OnboardingThemePicker,
-    AdapterItem.OnboardingTrackingProtection,
-    AdapterItem.OnboardingPrivateBrowsing,
-    AdapterItem.OnboardingPrivacyNotice,
-    AdapterItem.OnboardingFinish
-)
+private fun onboardingAdapterItems(onboardingState: OnboardingState): List<AdapterItem> {
+    val items: MutableList<AdapterItem> = mutableListOf(AdapterItem.OnboardingHeader)
+
+    // TODO customize the UI based on different account states.
```
Comment (sblatz) [inline app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlUIView.kt]: Ticket filed for this?

Comment (boek) [reply app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlUIView.kt]: Doh, fixed the todos, left the comments

**Pull request merged**